Algorithms/Ids.py

```diff
@@ -3,51 +3,56 @@
 from collections import deque
 from Util.utils import MoveTypes
 
-
-def Ids(board: Board, max_depth):
+def Ids(board: Board):
     layer = 0
 
     goal: list[int] = board.get_goal_state()
 
-    steps: list[string]
+    steps: list[MoveTypes]
 
     while True:
-        stack = deque()
-        stack.append(board)
-        visited = set()
-        steps = run_dfs_in_layer(stack, visited, goal, layer, [])
 
-        if steps is not None or layer > max_depth:
+        steps = run_dfs_in_layer(board, layer)
+
+        if steps is not None:
             break
         layer += 1
 
     if not steps:
```

Comment:
You arrived this condition if the steps is not None according to the while loop

Reply:
I think this is what you meant, but I am not sure